### PR TITLE
Upgrade dependecies 2

### PR DESCRIPTION
**Pull request by rahul-rathore-576**
### Before submitting the PR, please make sure you do the following

 1. Contributor license agreement
For us it's important to have the agreement of our contributors to use their work, whether it be code or documentation. Therefore, we are asking all contributors to sign a contributor license agreement (CLA) as commonly accepted in most open source projects. **Just open the pull request and our CLA bot will prompt you briefly.**

 2. Please check our [contribution guidelines](https://docs.traduora.com/docs/contributing#review-process-for-this-repo) for some help in the process.

### CODE REVIEW

**Comment by rahul-rathore-576** [comment]
@evereq 

Hi,
I have upgraded webapp (Angualar) and api (NestJs) application to latest version.

Bumps NestJs from 6.5.3 to 7.6.15.
Bumps Angular from 9.1.0 to 11.2.13.

Please let me know if you want any changes from my side.
Thank you!

File: api/src/main.ts [hunk 1]
```diff
@@ -46,8 +46,11 @@ async function bootstrap() {
       )
       .setVersion(version)
       .setBasePath('/')
-      .addOAuth2('password', '/api/v1/auth/token', '/api/v1/auth/token')
```
Comment (evereq) [inline api/src/main.ts:49]: @rahul-rathore-576 this seems to be some "change" in how it was before. Can't we preserve both parameters? Or they got same default values now?

File: api/src/formatters/yaml-nested.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { config } from '../config';
 import { Exporter, IntermediateTranslationFormat, Parser } from '../domain/formatters';
 
 export const yamlNestedParser: Parser = async (data: string) => {
-  const parsed = yaml.safeLoad(data);
+  const parsed = yaml.load(data);
```
Comment (evereq) [inline api/src/formatters/yaml-nested.ts:6]: @rahul-rathore-576 is there "safeLoad" method now or some parameter to load() which make it "safe"? Or it's safe by default now?

File: webapp/angular.json [hunk 3]
```diff
@@ -40,7 +43,6 @@
               "optimization": true,
               "outputHashing": "all",
               "sourceMap": false,
-              "extractCss": true,
```
Comment (evereq) [inline webapp/angular.json:43]: @rahul-rathore-576 why we removed that one?

File: webapp/src/polyfills.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 /***************************************************************************************************
  * Load `$localize` onto the global scope - used if i18n tags appear in Angular templates.
  */
-import '@angular/localize/init';
+// import '@angular/localize/init';
```
Comment (evereq) [inline webapp/src/polyfills.ts:4]: @rahul-rathore-576 if it's not relevant in new Angular version, let's just remove, no need to leave it commented out

File: webapp/src/tsconfig.app.json [hunk 5]
```diff
@@ -1,15 +1,18 @@
...
+	},
+	"files": [
+		"main.ts",
+		"polyfills.ts"
+	],
+	"include": [
+		"src/**/*.d.ts"
+	],
+	"angularCompilerOptions": {
+		"enableIvy": false
```
Comment (evereq) [inline webapp/src/tsconfig.app.json:16]: @rahul-rathore-576 did you try to enable Ivy? It did not work? Too many changes required? Asking not for this PR specifically (it's fine to merge with Ivy disabled), but for the future to understand how much effort we need to do to enable Ivy?

File: api/src/config.ts [hunk 6]
```diff
@@ -51,8 +51,8 @@ export const config = {
       host: env.TR_DB_HOST || '127.0.0.1',
       port: parseInt(env.TR_DB_PORT, 10) || 3306,
       username: env.TR_DB_USER || 'root',
-      password: env.TR_DB_PASSWORD || '',
-      database: env.TR_DB_DATABASE || 'tr_dev',
+      password: env.TR_DB_PASSWORD || 'root',
+      database: env.TR_DB_DATABASE || 'ever-traduora',
```
Comment (evereq) [inline api/src/config.ts:55]: @rahul-rathore-576 let's keep old default name for DB for now, for backward compatibility for users who don't submit own DB name etc. I.e. revert changes for both default password and DB name please

File: api/src/main.ts (hunk 1 above)
Comment (rahul-rathore-576) [reply api/src/main.ts:49]: @evereq I have updated `addOAuth2` method as per latest version of `@nestjs/swagger`. Can you please review it?

File: api/src/formatters/yaml-nested.ts (hunk 2 above)
Comment (rahul-rathore-576) [reply api/src/formatters/yaml-nested.ts:6]: @evereq In latest version of `js-yaml` they are completely removed `safeLoad` and `safeDump` methods and use now `load` and `dump` methods instead.

File: webapp/angular.json (hunk 3 above)
Comment (rahul-rathore-576) [reply webapp/angular.json:43]: @evereq Angular version (11.2.13) `ng update` CLI automatically removed `extractCss` option from angular.json by itself.

File: webapp/src/polyfills.ts (hunk 4 above)
Comment (rahul-rathore-576) [reply webapp/src/polyfills.ts]: @evereq I have removed `@angular/localize` from polyfills.ts.

File: webapp/src/tsconfig.app.json (hunk 5 above)
Comment (rahul-rathore-576) [reply webapp/src/tsconfig.app.json]: @evereq I have tried with enabled Ivy and its working fine.

File: api/src/config.ts (hunk 6 above)
Comment (rahul-rathore-576) [reply api/src/config.ts]: @evereq I have reverted my changes.

**Comment by evereq** [comment]
@rahul-rathore-576 please check 2 failing tests and try to fix that, see more details below (from our Slack)

We have 2 tests failing https://app.circleci.com/pipelines/github/ever-co/ever-traduora/463/workflows/40959840-10e4-4269-b156-c0c43551d307/jobs/515
The reason they are failing is unknown to me, i.e. not sure why.
But from what I see it takes json from simple-nested.json file 

```
"term:four": "hello there you\\nthis should be in a newline"
```
(simple-nested.yaml)

and try to compare it to what it should be in yaml file (simple-nested.yaml)

```
'term:four': hello there you\nthis should be in a newline
```

and somehow it fails
because it produces the wrong YAML from what I understand
i.e. it produce

```
term:four: hello there you\nthis should be in a newline
```
and you can see no ' symbol around the key value
i.e. should be 'term:four', but we get term:four.
(no ' symbol)

So we need to figure out why that happens and fix code the way that this test pass
:wink:

Btw, think you can run tests manually with
`yarn test && yarn test:e2e`

**Comment by evereq** [comment]
PR merged. Issue with tests was before all updates in this PR, so it will be fixed in some other branch / PR.

**Pull request merged**